src/test/librbd/test_librbd.cc

```diff
@@ -4232,6 +4232,68 @@ interval_set<uint64_t> round_diff_interval(const interval_set<uint64_t>& diff,
   return rounded_diff;
 }
 
+TEST_F(TestLibRBD, TestSizeImageSnap)
+{
+  rados_ioctx_t ioctx;
+  rados_ioctx_create(_cluster, m_pool_name.c_str(), &ioctx);
+
+  rbd_image_t image;
+  int order = 0;
+  std::string image_name = get_temp_image_name();
+  uint64_t size = 16 << 20;
+  uint64_t features;
+  features = (RBD_FEATURE_EXCLUSIVE_LOCK |
+              RBD_FEATURE_OBJECT_MAP |
+              RBD_FEATURE_FAST_DIFF);
+  ASSERT_EQ(0, create_image_full(ioctx, image_name.c_str(), size, &order,
+				 false, features));
+  ASSERT_EQ(0, rbd_open(ioctx, image_name.c_str(), &image, nullptr));
+
+  char test_data[TEST_IO_SIZE + 1];
+  for (size_t i = 0; i < TEST_IO_SIZE; ++i) {
+    test_data[i] = (char) (rand() % (126 - 33) + 33);
+  }
+  test_data[TEST_IO_SIZE] = '\0';
+
+  ASSERT_PASSED(write_test_data, image, test_data, 0,
+                TEST_IO_SIZE, LIBRADOS_OP_FLAG_FADVISE_NOCACHE);
+
+  interval_set<uint64_t> diff;
+  ASSERT_EQ(0, rbd_diff_iterate2(image, nullptr, 0, size, true, true,
+                                 iterate_cb, &diff));
+  ASSERT_EQ(diff.size(), 4194304);
```

Comment:
```suggestion
  EXPECT_EQ(diff.size(), 1 << order);
```

Reply:
Good catch, a hard coded value is less than ideal. I'll also change ordering of values compared to EXPECT_EQ(1 << order, diff.size()); to be consistent with other EQ with format of (expected value, variable)